vms/proposervm/vm_test.go

```diff
@@ -2530,3 +2531,43 @@ func TestLocalParse(t *testing.T) {
 		})
 	}
 }
+
+func TestTimestampMetrics(t *testing.T) {
+	ctx := context.Background()
+
+	coreVM, _, proVM, _ := initTestProposerVM(t, time.Unix(0, 0), mockable.MaxTime, 0)
+	defer func() {
+		require.NoError(t, proVM.Shutdown(ctx))
+	}()
+
+	innerBlock := snowmantest.BuildChild(snowmantest.Genesis)
+
+	outerTime := time.Unix(314159, 0)
+	innerTime := time.Unix(142857, 0)
+	proVM.Clock.Set(outerTime)
+	innerBlock.TimestampV = innerTime
+
+	coreVM.BuildBlockF = func(context.Context) (snowman.Block, error) {
+		return innerBlock, nil
+	}
+	outerBlock, err := proVM.BuildBlock(ctx)
+	require.NoError(t, err)
+	require.IsType(t, &postForkBlock{}, outerBlock)
+	require.NoError(t, outerBlock.Accept(ctx))
+
+	gaugeVec := proVM.lastAcceptedTimestampGaugeVec
+	tests := []struct {
+		blockType string
+		want      time.Time
+	}{
+		{innerBlockTypeMetricLabel, innerTime},
+		{outerBlockTypeMetricLabel, outerTime},
+	}
+	for _, tt := range tests {
```

Comment:
nit why `tt` and not `test` 👀 I suppose not to conflict with `t`? 🤔  But `test` is nice and short too 😅  Just a general comment for future code, please no need to change it here.

Reply:
Yeah `tt` to not shadow `t`. It was common enough in the Google code base (along with `tc` for test case) that it was generally known exactly what it was (like `ctx`) so I picked it up as a habit, but appreciate that that may not be the case outside. I tend to keep using it over `test` because of the number of times it gets used.

Reply:
I'd probably slightly prefer `test` as well - but I don't really care tbh